qa/tasks/cephfs/test_scrub.py

```diff
@@ -176,3 +176,8 @@ def test_scrub_backtrace(self):
 
     def test_scrub_dup_inode(self):
         self._scrub(DupInodeWorkload(self, self.fs, self.mount_a))
+
+    def test_mdsdir_scrub_backtrace(self):
```

Comment:
Does this test verify if scrub _fixes_ the damaged health status?

Reply:
IIRC - you had changes to trim entries from damage table. Did you forget to include those in the commit?

Reply:
> IIRC - you had changes to trim entries from damage table. Did you forget to include those in the commit?

I removed that change intentionally. As now we are skipping that case for mdsdir, I guess its not required then.

Reply:
> Does this test verify if scrub _fixes_ the damaged health status?

The `HEALTH_ERR` was due to the entry in damage table. If we ensure that after running scrub , there's no increase in damage count. Means, the entry is not recorded in damage table and hence it's now not responsible for any `HEALTH_ERR`.

Reply:
@neesingh-rh it would be good to explain something like above in the commit too, would make it more understandable.

Reply:
> > Does this test verify if scrub _fixes_ the damaged health status?
> 
> The `HEALTH_ERR` was due to the entry in damage table. If we ensure that after running scrub , there's no increase in damage count. Means, the entry is not recorded in damage table and hence it's now not responsible for any `HEALTH_ERR`.

But that is _not_ the intention of this fix, isn't it? The issue is that scrub does not clear damage health status. You mention that the damage status is not recorded since there is no damage detected during scrub. So, how do you verify the fix?

Reply:
i understand what venky is saying, this patch does fix the issue but does't address the title of the PR. Maybe commit title should be "do not add entry in damage table in case of mdsdir inode" or something like this. and as venky said, we need some verification and that can be done by checking the damage table entry count pre and post scrubbing. What do you think?

Reply:
I have included the checks for the `health_status` too in the test for verification.

Reply:
maybe this sounds better: **mds: avoid scrubbing and reporting damage status for ~mdsdir**

Reply:
Exactly. The PR description and the fix were not making sense.

@neesingh-rh I presume you'll be sending a fix for the actual tracker issue then? (inodes other than ~mdsdir can be damaged and we want scrub to clear the health status once the damage is fixed).

BTW, this fix needs a separate tracker.

Reply:
> @neesingh-rh I presume you'll be sending a fix for the actual tracker issue then? (inodes other than ~mdsdir can be damaged and we want scrub to clear the health status once the damage is fixed).
> 
@vshankar I could only reproduce this issue with ~mdsdir not in other cases.And as now we won't be encountering with mdsdir too. I don't think that is a problem now.

Reply:
You'd need to explicitly damage an inode's metadata and then verify that scrub clears damage health status. look at test_damage.py.

Reply:
@vshankar Now, I understood after damaging an inode's metadata explicitly. After running `scrub start` on the inode , we are getting entry in damage table and even after scrub repair , it not getting removed from damage table and hence the health status too.But seeing the logs, it is repairing the inode and not finding it in next run. It means I should keep my change of trimming the entries from damage table. I will create another PR for that and link this PR to another new tracker. Thanks:)

Reply:
@vshankar PTAL